lib/lints/avoid_late_keyword/avoid_late_keyword_rule.dart

```diff
@@ -31,9 +34,29 @@ class AvoidLateKeywordRule extends SolidLintRule {
     CustomLintContext context,
   ) {
     context.registry.addVariableDeclaration((node) {
-      if (node.declaredElement?.isLate ?? false) {
+      if (_shouldLint(node)) {
         reporter.reportErrorForNode(code, node);
       }
     });
   }
+
+  bool _shouldLint(VariableDeclaration node) {
+    final isLateDeclaration = node.declaredElement?.isLate ?? false;
+    if (!isLateDeclaration) return false;
+
+    final ignoredTypes = _hasIgnoredType(node);
+    if (ignoredTypes) return false;
+
+    final allowInitialized = config.parameters.allowInitialized;
+    if (!allowInitialized) return true;
+
+    final hasInitializer = node.initializer != null;
+    return !hasInitializer;
+  }
+
+  bool _hasIgnoredType(VariableDeclaration node) =>
+      config.parameters.ignoredTypes.contains(
+        // ignore: deprecated_member_use
+        node.declaredElement2?.type.getDisplayString(withNullability: false),
```

Comment:
Is there a way to solve this without using deprecated APIs?
Also, this checks for exact match, not subtypes.